### PR TITLE
Persist claim issuers in server side sessions

### LINKED ISSUE #1546: Claim Issuer not persisted when using Server Side Sessions
**Which version of Duende IdentityServer are you using?**
7

**Which version of .NET are you using?**

8

**Describe the bug**

When using Server Side Sessions, the issuer of the claim is not persisted, which means on rehydration the issuer is set to `LOCAL_AUTHORITY` which is the default for a claim.

This causes issues with the Sustainsys.SAML2 Single Logout as it uses the issuer from the claim in order to redirect to the identity provider. See https://github.com/Sustainsys/Saml2/blob/3780695449868fdeb708f0665eadd042a1a27153/Sustainsys.Saml2/WebSSO/LogOutCommand.cs#L168

**To Reproduce**

Setup identity server that uses Sustainsys.SAML2 as an external identity provider with single logout enabled and enable server side sessions.
Then attempt to logout.

**Expected behavior**

The claim should be rehydrated with the original issuer in order to allow single logout.

--

The question is, was the omission of persisting the issuer a design choice and this should be handled in a different manner?

### PR DESCRIPTION
**Pull request by josephdecock**
Resolves #1546

@AndersAbel this came up in context of a support issue you worked where someone couldn't logout using saml and sustainsys, so I've added you as a reviewer if you have time - is this what you were hoping for?

### CODE REVIEW

File: src/Storage/Stores/Serialization/ClaimLite.cs [hunk 1]
```diff
@@ -11,4 +11,5 @@ public class ClaimLite
     public string Type { get; set; }
     public string Value { get; set; }
     public string ValueType { get; set; }
-}
\ No newline at end of file
+    public string Issuer { get; set; }
```
Comment (damianh) [inline src/Storage/Stores/Serialization/ClaimLite.cs:14]: Any concerns about about existing valves being null ?

Comment (josephdecock) [reply src/Storage/Stores/Serialization/ClaimLite.cs:14]: Nullable reference types aren't enabled in this file, but Issuer and ValueType both in practice can be null and we handle them similarly when we deserialize.

**Review by damianh — APPROVED** [review]
Looks good to me. But also wait for Anders

**Comment by AndersAbel** [comment]
This should work, but it would be possible to improve the optimization for the default issuer.

All claims in a ClaimsIdentity from a remote authentication provider typically have the same issuer. For Saml2 it's the Entity Id of the Idp (e.g. "https://stubidp.sustainsys.com/Metadata"). For OIDC it's the issuer/authority (e.g. "https://demo.duendesoftware.com").

For storage in a DB I don't think it's an issue to store the same string for each claim. As far as I remember this code would not be used to store anything in a cookie. If that was the case. I would have tried to optimize further to keep the issuer string once somewhere.

**Pull request merged**